torch/utils/flop_counter.py

```diff
@@ -722,8 +723,12 @@ def __exit__(self, *args):
 
     def __torch_dispatch__(self, func, types, args=(), kwargs=None):
         kwargs = kwargs if kwargs else {}
-        out = func(*args, **kwargs)
-        return self._count_flops(func._overloadpacket, out, args, kwargs)
+        if func._can_decompose():
+            with self.decomposed_counter:
```

Comment:
I think you don't need a separate dedicated mode, and you can just say `with self` here?

Reply:
The issue is that entering `self` a second time resets the counter's state. There are two possible solutions:
- A second context manager that does not reset the counter's state, as my code does
- A flag that is set before re-entering the context manager, which is checked in `__enter__` to not reset the state

I'm not sure what's the better solution. What do you think?

Reply:
Oh oops, I see the reset. The flopcounter shouldn't have been implemented this way. But yes, we will need an inner manager. Actually, if you're feeling adventurous, you could refactor the *entire* dispatch mode into its own dedicated private class, I'd like that better than duping it as is here